light/rpc/client.go

```diff
@@ -340,7 +340,7 @@ func (c *Client) BlockResults(height *int64) (*ctypes.ResultBlockResults, error)
 
 	// Verify block results.
 	results := types.NewResults(res.TxsResults)
-	if rH, tH := results.Hash(), trustedHeader.LastResultsHash; !bytes.Equal(rH, tH) {
+	if rH, tH := results.Hash(), trustedHeader.LastResultsHash; len(rH) != 0 && !bytes.Equal(rH, tH) {
```

Comment:
I think this byte-comparison is wrong because `LastResultsHash` is calculated by building a merkle tree out of the above 3 binary as following https://github.com/tendermint/tendermint/blob/0b1722d78e977521ed332cf0c85b3b9251f9bb48/state/store.go#L299

And you must check whether `LastResultsHash` is empty result or not if `results.Hash()` is empty.

Reply:
I see, so do you think I should add `bbeBytes` and `ebeBytes` to `results.Hash()` and then compare it with `trustedHeader.LastResultsHash`